### ISSUE TITLE
multinode mpi application crashes with -mca btl vader,self,openib flag 

### ISSUE THREAD

**Issue by puneet336** [open-mpi/ompi#7555]
### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)
OpenMPI Version - 4.0.x (tried 4.0.3 and 4.0.0)

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
openmpi was installed from sources using - 
./configure --prefix=$OPENMPIROOT  CC=${CC} CXX=${CXX} FC=${FC} CFLAGS="-O3 -march=znver2 -fPIC" CXXFLAGS="-O3 -fPIC"  FCFLAGS="-O3  -fPIC" --enable-mpi-fortran --enable-shared=yes --enable-static=yes  --enable-mpi1-compatibility --disable-hwloc-pci 2>&1|tee config_logs

### Please describe the system on which you are running

* Operating system/version:  RHEL 7.6
* Computer hardware:  AMD 7742
* Network type:  mellanox infiniband

-----------------------------

## Details of the problem

when i launch my job using -mca btl vader,self,openib or -mca btl vader,self
```shell
shell$ export OMPI_MCA_btl_openib_allow_ib=1
shell$ export OMPI_MCA_btl_openib_if_include="mlx5_1:1"
shell$ mpirun -np 32 -npernode 16   -mca btl vader,self,openib  ./gmx

--------------------------------------------------------------------------
At least one pair of MPI processes are unable to reach each other for
MPI communications.  This means that no Open MPI device has indicated
that it can be used to communicate between these processes.  This is
an error; Open MPI requires that all MPI processes be able to reach
each other.  This error can sometimes be the result of forgetting to
specify the "self" BTL.

  Process 1 ([[21527,1],0]) is on host: node001
  Process 2 ([[21527,1],128]) is on host: node002
  BTLs attempted: self vader

Your MPI job is now going to abort; sorry.
--------------------------------------------------------------------------
[node05:57512] *** An error occurred in MPI_Bcast
[node05:57512] *** reported by process [1410793473,0]
[node05:57512] *** on communicator MPI_COMM_WORLD
[node05:57512] *** MPI_ERR_INTERN: internal error
[node05:57512] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[node05:57512] ***    and potentially your MPI job)
[node05:57496] 127 more processes have sent help message help-mca-bml-r2.txt / unreachable proc
[node05:57496] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
[node05:57496] 127 more processes have sent help message help-mpi-errors.txt / mpi_errors_are_fatal
```
a) tried removing the -mca btl vader,self,openib flag as - 
```shell
shell$ export OMPI_MCA_btl_openib_allow_ib=1
shell$ export OMPI_MCA_btl_openib_if_include="mlx5_1:1"
shell$ mpirun -np 32 -npernode 16   ./gmx
```
b) used  -mca btl vader,self,tcp  as 
```shell
shell$ export OMPI_MCA_btl_openib_allow_ib=1
shell$ export OMPI_MCA_btl_openib_if_include="mlx5_1:1"
shell$ mpirun -np 32 -npernode 16   -mca btl vader,self,tcp  ./gmx``
```
 with both a) and b)  settings the application runs fine but i guess i am losing out on some optimizations with these settings as performance gets slower.

please let me know in case more information is required from my end.

**Comment by ggouaillardet** [comment]
It seems Open MPI falls back to tcp because openib is not working.
You can
```
mpirun --mca bth_base_verbose 10 ...
```
To confirm that

You can also
`ibstat`
on both nodes to check link is `ACTIVE` and both `mlx5_1:1` are in the same subnet.

**Comment by jsquyres** [comment]
You might want to switch over to using UCX (instead of openib).  That's the preferred way to use InfiniBand these days.  Indeed, the openib BTL has been removed from Open MPI's master development, meaning that it will not be included in Open MPI v5.0.x.

Can you build Open MPI with UCX support?

**Comment by puneet336** [comment]
yes, it was probably issue with the openib compatibility with HDR100 based clusters.  EDR based clusters, the same mpirun worked fine.
With UCX support enabled, i am able to spawn my executables on multiple machines.
closing this issue.
